test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_handlers_test.clj

```diff
@@ -171,6 +173,41 @@
           (is (= (:status response) 200))
           (is (= (:body response) init-param-two)))))))
 
+(deftest websocket-test
+  (testing "Websocket handlers"
+    (with-app-with-config app
+      [jetty9-service]
+      jetty-plaintext-config
+      (let [s                     (get-service app :WebserverService)
+            add-websocket-handler (partial add-websocket-handler s)
+            path                  "/test"
+            connected             (atom 0)
+            server-messages       (atom [])
+            client-messages       (atom [])
+            handlers              {:on-connect (fn [ws]
+                                                 (ws-session/send! ws "Hello client!")
+                                                 (swap! connected inc))
+                                   :on-text    (fn [ws text]
+                                                 (ws-session/send! ws (str "You said: " text))
+                                                 (swap! server-messages conj text))}]
```

Comment:
Minor nitpicks w/rt testing:

1. It'd probably be good to have a test that exercised each one of the different types of handlers if at all possible.  If we were able to flag this stuff as `experimental` in an obvious way, as I mentioned in my earlier comment, then I might be satisfied with putting some kind of TODO comment here that says "we need to add tests for `on-close`, ... as soon as we move this out of "experimental".
2. I'm not sure how much testing we do for other types of handlers on the webrouting side of things, but it seems like we might need at least a really basic test for that.  @fpringvaldsen what do you think?